lib/posix-socket/socket_vnops.c

```diff
@@ -290,7 +290,7 @@ posix_socket_vfscore_ioctl(struct vnode *vnode,
 		ret = -ret;
 	}
 
-	return 0;
+	return ret;
```

Comment:
It only makes sense to allow returning a positive non-zero value from `posix_socket_ioctl` (no error case) if you want to conform to the `ioctl` system call, which says that an ioctl may return a positive value. This would be nice to have, but since vfscore is an absolute mess, it expects positive error values. This is why the return value is negated on error. So, returning a positive non-zero value from `posix_socket_ioctl` is not possible due to the interpretation by vfscore as error value. The function thus simply returns 0. I don't see a reason to change that behavior.

However, I agree, that there is a bug in that line `290` should be `return -ret`.

Reply:
Okay, you are right. To prevent unnecessary merge conflicts I'll fix that as part of the fdtab refactoring.

Reply:
Merge conflicts?

Reply:
I did quite some extensive changes to vfscore. If I fix the error handling now, I'll have to update the changes I did for fdtab.

Reply:
Hmm, you probably have to solve merge conflicts / rebase anyways, when you prepare your fdtab work for upstream. Since we have the discussion here, I think it would be better to have it fixed with this PR.

Reply:
Okay, fine by me :+1: